File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.js
const prouns = ['aya', 'ye', 'woo', 'ma'];

function solution(babbling) {
  console.log(babbling);
    
    const filtered = babbling.filter(b => {
        if(/^(?:aya|ye|woo|ma)+$/.test(b)){
            
            if(!(/ayaaya|yeye|woowoo|mama/).test(b)){
                return true;    
            } else {
                return false
            }
        } else {
            return false;
        }
    })
    
    console.log(filtered.length)
    
    return filtered.length;
}

File: 프로그래머스/1/133499. 옹알이 （2）/옹알이 （2）.py
import re

def solution(babbling):
    prounce = ['aya', 'ye', 'woo', 'ma']
    p_pipe = "|".join(prounce)
    
    regex = rf"^(?!.*({p_pipe})(?=\1))(?:{p_pipe})+$"
    
    return sum(1 for b in babbling if re.match(regex, b))
        